SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1295,7 +1307,58 @@ void USpatialActorChannel::ResetShadowData(FRepLayout& RepLayout, FRepStateStati
 	}
 }
 
-bool USpatialActorChannel::SatisfiesSpatialPositionUpdateRequirements(FVector& OutNewSpatialPosition)
+void USpatialActorChannel::CheckForClientEntityInterestUpdate()
+{
+	// Only valid if rep graph enabled, client entity interest is enabled, and we're processing a player controller
+	const double CurrentTime = NetDriver->GetElapsedTime();
+	const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>();
+	const UReplicationGraph* RepGraph = Cast<USpatialReplicationGraph>(NetDriver->GetReplicationDriver());
+	USpatialNetConnection* NetConnection = Cast<USpatialNetConnection>(Actor->GetNetConnection());
+	UNetReplicationGraphConnection* RepGraphConnection =
+		Cast<UNetReplicationGraphConnection>(NetConnection->GetReplicationConnectionDriver());
+
+	if (RepGraph == nullptr || RepGraphConnection == nullptr || NetConnection == nullptr)
+	{
+		UE_LOG(LogSpatialActorChannel, Error,
+			   TEXT("Failed to handle client entity interest, some connection or setting was misconfigured"));
+		return;
+	}
+
+	// If interest is already marked dirty (e.g. because we just gained authority over the PC and want to flush immediate),
+	// then don't bother doing any checks.
+	if (NetDriver->ActorSystem->IsClientInterestDirty(EntityId))
+	{
+		NetConnection->TimeWhenClientInterestLastUpdated = CurrentTime;
+		return;
+	}
+
+	bool bShouldMarkInterestDirty = false;
+
+	// Round robin updating client interest
+	if (RepGraph->GetReplicationGraphFrame() % Settings->ClientEntityIdInterestUpdateFrameFrequency
```

Comment:
Nit: This might have been done for clarity in which case coool! Otherwise bShouldMarkInterestDirty seems unnecessary. We could check against the condition and return if false.

Reply:
Yeah just for clarity + debugging